### PR TITLE
Display Header with Flexbox

### PR DESCRIPTION
**Pull request by pistou**
Remove the use of `position: absolute` and use `Flexbox` instead.

I have no idea why it used to be *absolute positioned* but that made it hard to easily modify, as proven by [Issue#196](https://github.com/react-navigation/stack/issues/196).

This PR should fix this kind of issue, as such:
```
class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Home',
    headerTitleContainerStyle: {
      alignItems: 'center', // center horizontally
    },
  };

  render() {
    return (
      <View>
        <Text>Hello World!</Text>
      </View>
    );
  }
}
```

*I use only Android, I couldn't test on iOS but it **should** work the same way*

### CODE REVIEW

File: src/views/Header/HeaderSegment.tsx [hunk 1]
```diff
@@ -331,7 +331,7 @@ export default class HeaderSegment extends React.Component<Props, State> {
               style={[
                 Platform.select({
                   ios: null,
-                  default: { left: leftButton ? 72 : 16 },
+                  default: { paddingLeft: leftButton ? 0 : 16 },
```
Comment (pistou) [inline src/views/Header/HeaderSegment.tsx]: We don't need a padding if there is a `leftButton` as the `title container` will be pushed by the button itself

**Comment by satya164** [comment]
Can you post screenshots and gifs on both Android and iOS after your change so we are sure that there's no regression?

**Comment by pistou** [comment]
As said, I don't have an iOS hardware to test it, only Android.

Here is a comparison of before / after:
before---after


GIF before:
old

GIF after:
new

**Comment by satya164** [comment]
> I use only Android, I couldn't test on iOS but it should work the same way

On iOS, the back button has a back label enabled by default and the title is centered. Pretty sure that it's not possible to emulate that using flexbox.

> I don't have an iOS hardware to test it

You don't need iOS. You can test in a simulator on a Mac. If you don't have a Mac, you can try switching the `Platform.OS` checks to see how it might look on iOS.

**Comment by satya164** [comment]
Closing due to no response

**Pull request closed without merging**